smart-contracts/contracts/Unlock.sol

```diff
@@ -160,8 +231,13 @@ contract Unlock is
       mstore(add(pointer, 0x14), _salt)
       salt := mload(pointer)
     }
-    address payable newLock = payable(address(uint160(publicLockAddress.createClone2(salt))));
-    IPublicLock(newLock).initialize(
+
+    // default to latest implementation
+    address impl = _publicLockImpls[publicLockLatestVersion];
+
+    // TODO: more flexible approche to init w params
+    bytes memory data = abi.encodeWithSignature(
+      'initialize(address,uint256,address,uint256,uint256,string)',
```

Comment:
same as above. we need to pass a `data` blob that matches the actual `initialize` call + its params (the same way the DAO proposals are built I guess!)